### ISSUE TITLE
Drop support for Python 3.4?

### ISSUE THREAD

**Issue by hugovk** [python-pillow/Pillow#3581]
Python 3.4 reaches [EOL on 2019-03-16](https://devguide.python.org/#status-of-python-branches) and is no longer supported by the core Python team.

I propose the 2019-04-01 release (or another TBD) should be major release 6.0.0 and we also bundle in some deprecation warning removals at the same time.

* [x] Drop Python 3.4
   * PRs: https://github.com/python-pillow/Pillow/pull/3596, https://github.com/python-pillow/pillow-wheels/pull/113, https://github.com/python-pillow/docker-images/pull/53

* From the [deprecations page](https://pillow.readthedocs.io/en/stable/deprecations.html), but only if deprecated for a year (per https://github.com/python-pillow/Pillow/issues/1087#issuecomment-121655443):

  * [x] Remove deprecated (since 4.0.0/2017-01) PIL.OleFileIO in favour of olefile Python package (https://github.com/python-pillow/Pillow/pull/2199, https://github.com/python-pillow/Pillow/pull/2784)
      * PR: https://github.com/python-pillow/Pillow/pull/3598

  * [x] Remove deprecated (4.3.0/2017-10) ImageOps undocumented functions (https://github.com/python-pillow/Pillow/pull/2735)
      * PR: https://github.com/python-pillow/Pillow/pull/3599

  * [x] Remove deprecated (5.2.0/2018-07) ~`PILLOW_VERSION` and~ `VERSION` (https://github.com/python-pillow/Pillow/pull/3090)
      * PR: https://github.com/python-pillow/Pillow/pull/3624

  * ~Remove deprecated (5.3.0/2018-10) setting of TIFF image sizes, make read-only (https://github.com/python-pillow/Pillow/pull/3203)~ (<1 year)

  * [x] Docs and release notes: https://github.com/python-pillow/Pillow/pull/3600

* Anything else?

Thoughts, objections, additions?

If 2019-04-01 is too early, when would be a suitable time?

**Comment by hugovk** [comment]
@python-pillow/pillow-team: Feedback welcome!

**Comment by hugovk** [comment]
# VERSION deprecation

Added in PR https://github.com/python-pillow/Pillow/pull/3090, released in 5.2.0 on 2018-07-01, `__init__.py` and `Image.py` say:

```python
# VERSION is deprecated and will be removed in Pillow 6.0.0.
# PILLOW_VERSION is deprecated and will be removed after that.
# Use __version__ instead.
VERSION = '1.1.7'  # PIL Version
PILLOW_VERSION = __version__ = _version.__version__
```

Also in release notes:
* https://pillow.readthedocs.io/en/stable/releasenotes/5.2.0.html#deprecations

And recently added on this new page:
* https://pillow.readthedocs.io/en/stable/deprecations.html

6.0.0 will be out 9 months after 5.2.0.

Shall we press ahead and remove `VERSION` in the upcoming 6.0.0, as advertised? (`PILLOW_VERSION` will remain until a later release.)

Or give it some more time and postpone it to 7.0.0?

**Comment by hugovk** [comment]
The static `VERSION` (1.1.7) is the least useful of the two, so I think we're fine removing it in 6.0.0. See PR https://github.com/python-pillow/Pillow/pull/3624.
